tests/meta/workflow/test_analysis.py

```diff
@@ -32,15 +34,25 @@
         (Priority.express, SlurmQos.EXPRESS),
     ],
 )
-def test_get_slurm_qos_for_case(mocker, case_id: str, priority, expected_slurm_qos):
-    """Test qet Quality of service (SLURM QOS) from the case priority"""
+def test_get_slurm_qos_for_case(
+    case_id: str,
+    priority,
+    expected_slurm_qos,
+    mip_analysis_api: MipDNAAnalysisAPI,
+    analysis_store: Store,
+):
+    """Test get Quality of service (SLURM QOS) from the case priority"""
 
-    # GIVEN a case that has a priority
-    mocker.patch.object(AnalysisAPI, "get_priority_for_case")
-    AnalysisAPI.get_priority_for_case.return_value = priority
+    # GIVEN a store with a case with a specific priority
+    case: Case = analysis_store.get_case_by_internal_id(case_id)
+    case.priority = priority
+
+    with mock.patch.object(
+        mip_analysis_api.status_db, "get_case_by_internal_id", return_value=case
+    ):
```

Comment:
You should be able to skip the mocking by just setting `mip_analysis_api.status_db = analysis_store`. I do not see why we would mix both mocking statusdb but also having a functioning and sufficient version of it.

Reply:
ok yes I agree 👍